ClosedXML/Excel/CalcEngine/Expression.cs

```diff
@@ -509,10 +509,14 @@ public override object Evaluate()
 
         public IEnumerator GetEnumerator()
         {
-            if (_value is string)
-                return new[] { (string)_value }.GetEnumerator();
+            if (_value is string s)
+                yield return s;
 
-            return (_value as IEnumerable).GetEnumerator();
+            if (_value is IEnumerable ie)
```

Comment:
Here we should use `else if`, otherwise we'll return a string and _then_ all chars it consists of. A known gotcha of `yield return` ;)